src/librustc/session/config.rs

```diff
@@ -1285,6 +1283,95 @@ pub fn parse_crate_types_from_list(list_list: Vec<String>) -> Result<Vec<CrateTy
     return Ok(crate_types);
 }
 
+pub struct NightlyOptions {
+    is_nightly_build: bool,
+    is_unstable_enabled: bool,
+    flags: Vec<RustcOptGroup>,
+}
+
+impl NightlyOptions {
+    pub fn get(matches: &getopts::Matches) -> NightlyOptions {
+        let nightly_build = get_unstable_features_setting();
+        NightlyOptions {
+            is_nightly_build: match nightly_build {
+                                  UnstableFeatures::Allow | UnstableFeatures::Cheat => true,
+                                  _ => false,
+                              },
+            is_unstable_enabled: matches.opt_strs("Z").iter().any(|x| *x == "unstable-options"),
+            flags: rustc_optgroups(),
```

Comment:
I think this needs to be passed in because it would otherwise be wrong for rustdoc